doc/rados/operations/crush-map-edits.rst

```diff
@@ -240,121 +247,136 @@ and two node buckets named ``host`` and ``rack`` respectively.
-	   or re-weighting of an item. The straw bucket type allows all items to
-	   fairly “compete” against each other for replica placement through a
-	   process analogous to a draw of straws.
-
-        #. **straw2**: Straw2 buckets improve Straw to correctly avoid any data
-           movement between items when neighbor weights change.
-
-           For example the weight of item A including adding it anew or removing
-           it completely, there will be data movement only to or from item A.
+   Ceph supports five bucket types. Each bucket type provides a balance between
+   performance and reorganization efficiency, and each is different from the
+   others.  If you are unsure of which bucket type to use, use the ``straw2``
+   bucket. For a more technical discussion of bucket types than is offered
+   here, see **Section 3.4** of `CRUSH - Controlled, Scalable, Decentralized
+   Placement of Replicated Data`_. 
+   
+   The bucket types are as follows:
+
+    #. **uniform**: Uniform buckets aggregate devices that have **exactly** 
+       the same weight. For example, when hardware is commissioned or
+       decommissioned, it is often done in sets of machines that have exactly
+       the same physical configuration (this can be the case, for example,
+       after bulk purchases).  When storage devices have exactly the same
+       weight, you may use the ``uniform`` bucket type, which allows CRUSH to
+       map replicas into uniform buckets in constant time. If your devices have
+       non-uniform weights, you should not use the uniform bucket algorithm.
+
+    #. **list**: List buckets aggregate their content as linked lists. The 
+       behavior of list buckets is governed by the :abbr:`RUSH (Replication
+       Under Scalable Hashing)`:sub:`P` algorithm. In the behavior of this
+       bucket type, an object is either relocated to the newest device in
+       accordance with an appropriate probability, or it remains on the older
+       devices as before. This results in optimal data migration when items are
+       added to the bucket. The removal of items from the middle or the tail of
+       the list, however, can result in a significant amount of unnecessary
+       data movement. This means that list buckets are most suitable for
+       circumstances in which they **never shrink or very rarely shrink**.
+
+    #. **tree**: Tree buckets use a binary search tree. They are more efficient
+       at dealing with buckets that contain many items than are list buckets.
+       The behavior of tree buckets is governed by the :abbr:`RUSH (Replication
+       Under Scalable Hashing)`:sub:`R` algorithm.  Tree buckets reduce the
+       placement time to 0(log\ :sub:`n`). This means that tree buckets are
+       suitable for managing large sets of devices or nested buckets.
+
+    #. **straw**: Straw buckets allow all items in the bucket to "compete" 
+       against each other for replica placement through an aleatory process
```

Comment:
I suspect that even readers whose first language is English will not know "aleatory"

Reply:
This is entirely my fault. "Alea iacta est" is never far from my mind, and I wrongly assume that that must be the case for everyone else as well.

I'll remove it, with regret (because it is exactly the right word).